Fix invalid prop type for isValidSection warning

Pytal opened the pull request:
Fix invalid prop type for isValidSection warning

```
[Vue warn]: Invalid prop: type check failed for prop "isValidSection". Expected Boolean, got String with value "English (US)".
```

Pytal committed:
Fix invalid prop type for isValidSection

Signed-off-by: Christopher Ng <chrng8@gmail.com>
Signed-off-by: nextcloud-command <nextcloud-command@users.noreply.github.com>

Pytal commented:
/compile amend /

kesselb commented on apps/settings/src/utils/validate.js:
```diff
@@ -64,5 +64,5 @@ export function validateEmail(input) {
 export function validateLanguage(input) {
 	return input.code !== ''
-		&& input.name
+		&& Boolean(input.name)
```
Isn't that `!== ''` just harder to read :thinking:

oltl4ever81 replied:
-

Pytal replied:
The reasoning was that input.code would always be guaranteed to be a string so the check is explicitly for a string, while name uses a property access here https://github.com/nextcloud/server/blob/fix/boolean-expected/apps/settings/src/components/PersonalInfo/LanguageSection/Language.vue#L125 which may be undefined :)

kesselb replied:
Thanks and good to know ;)

julien-nc reviewed (commented):
Works fine.

We could fix the jsdoc for `validateLanguage`, it seems the `input` param is not a String but an Object.

Wouldn't it be more explicit to replace the condition by the following one?
``` js
return input.code !== '' && input.name !== undefined && input.name !== ''
```

szaimen commented:
Is there a testing procedure? I'd like to test this but don't know how. Can you write some steps?

eneiluj commented:
> Is there a testing procedure? I'd like to test this but don't know how. Can you write some steps?

@szaimen To witness the warning, with the master branch, you can build the dev target for the settings module. Vue warnings don't show up if you execute the `build` target:

``` bash
cd /var/www/.../nextcloud.dev.git/
git pull origin master
git checkout master
# optionally make sure js packages are up to date
MODULE=settings npm ci
MODULE=settings npm run dev
```
Then load the personal user settings page, you should see the warning in the browser console log.

Then use this PR's branch, build again and reload the settings:

``` bash
git checkout fix/boolean-expected
MODULE=settings npm run dev
```

The warning should be gone and the language settings validation should still work fine.
The list of available JS modules can be found in `/webpack.common.js`.

szaimen commented:
Thanks for the detailed instructions but the compiling fails for me, unfortunatley. 
I will not investigate this though as it is currently not on my priority list but maybe at some point in the future :)

Pytal commented:
/compile amend /

Pytal commented:
/compile amend /

Pytal commented:
> We could fix the jsdoc for `validateLanguage`, it seems the `input` param is not a String but an Object.

Nice catch, fixed!

> Wouldn't it be more explicit to replace the condition by the following one?
> 
> ```js
> return input.code !== '' && input.name !== undefined && input.name !== ''
> ```
Updated :)

Pytal commented:
/compile amend /

julien-nc reviewed (approved):
:+1:

szaimen commented:
I think this branch needs a rebase to get rid of the conflicts...

Pytal commented:
/compile amend /
